### PR TITLE
lib/vfscore: Prevent superfluous slashes in working directory

### PR DESCRIPTION
**Pull request by jschlumpp**
The mount point can contain a trailing slash (for example the root mount point has a path of "/"). The path that is concatenated to the mount point also contains a slash at the beginning. Therefore, extra care must be taken to prevent two slashes from appearing besides each other.

### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [x] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): N/A

### CODE REVIEW

File: lib/vfscore/syscalls.c [hunk 1]
```diff
@@ -1216,7 +1218,14 @@ sys_fchdir(struct vfscore_file *fp, char *cwd)
 		vn_unlock(dvp);
 		return EBADF;
 	}
-	strlcpy(cwd, fp->f_dentry->d_mount->m_path, PATH_MAX);
+
+	// Cut of the last trailing slash if there is one
```
Comment (RaduNichita) [inline lib/vfscore/syscalls.c]:
```suggestion
	// Cut off the last trailing slash if there is one
```

**Review by RaduNichita — CHANGES_REQUESTED** [review]

**Review by StefanJum — APPROVED** [review]
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by RaduNichita — APPROVED** [review]
Looks good.

Reviewed-by: Radu Nichita <radunichita99@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**